manifests/repo/debian.pp

```diff
@@ -35,22 +35,22 @@
   Boolean $dotdeb         = true,
   Boolean $sury           = true,
 ) {
-  assert_private()
-
   include 'apt'
 
-  apt::source { "source_php_${release}":
-    location => $location,
-    release  => $release,
-    repos    => $repos,
-    include  => {
-      'src' => $include_src,
-      'deb' => true,
-    },
-    key      => $key,
+  if ($dotdeb) {
+    apt::source { "source_php_${release}":
+      location => $location,
+      release  => $release,
+      repos    => $repos,
+      include  => {
+        'src' => $include_src,
+        'deb' => true,
+      },
+      key      => $key,
+    }
   }
 
-  if ($sury and $php::globals::php_version in ['7.1','7.2']) {
```

Comment:
I believe here we should add 7.4 (and possible 7.3) instead if removing it.

Reply:
I thought it would be less troublesome to assume that the requested version was supported by Sury, rather than keeping an eye on them and patch puppet-php each time they release a new version (they must be shipping 8.0, or maybe even 8.1).